scipy/stats/_continuous_distns.py

```diff
@@ -5342,10 +5342,17 @@ def _munp(self, n, dfn, dfd, nc):
         return val
 
     def _stats(self, dfn, dfd, nc):
-        mu = np.where(dfd <= 2, np.inf, dfd / (dfd-2.0)*(1+nc*1.0/dfn))
-        mu2 = np.where(dfd <= 4, np.inf, 2*(dfd*1.0/dfn)**2.0 *
-                       ((dfn+nc/2.0)**2.0 + (dfn+nc)*(dfd-2.0)) /
-                       ((dfd-2.0)**2.0 * (dfd-4.0)))
+        mu = _lazywhere(dfd > 2,
+                        (dfn, dfd, nc),
+                        lambda dfn, dfd, nc: dfd / (dfd - 2) * (1 + nc/dfn),
+                        fillvalue=np.inf)
```

Comment:
Any risk that using `_lazywhere` with a `lambda` will end up being slower for large arrays because it will be called many times rather than being computed in NumPy (i.e., excessive Python function call overhead)?

Reply:
Yes, there is a cost to using `_lazywhere`, because it will generally require more copying than if it is not used.  We currently pay that cost in many places.  If you want to think about avoiding `_lazywhere`, it would be nice to tackle all those, too.

In some cases (such as here), it might be possible to replace `_lazywhere` with the judicious use of the `where` argument of numpy ufuncs.  I'll experiment a bit.